### PR TITLE
PACA-555: Resolve illegal reflection warning from gRPC module

### PR DESCRIPTION
**Pull request by KyRobbins**
**JIRA Ticket:**
[PACA-555](https://jira.cms.gov/browse/PACA-555)

**User Story or Bug Summary:**

---

### What Does This PR Do?

Resolves the "illegal reflection" warning logged by JDK9+ in the `bfd-pipeline-rda-grpc` module by adding the `mockito-inline` dependency along side the existing `mockito-core` dependency.

`mockito-inline` simply adds in a single line configuration file that configures Mockito to use the "inline" mock maker implementation, which unlike the default implementation, is able to mock final classes, protected fields, etc., without causing an illegal reflection warning.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [x] Yes
      * [ ] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [x] Yes
      * [ ] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by brianburton — DISMISSED** [review]

**Review by ericbuckley — DISMISSED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml [hunk 1]
```diff
@@ -168,6 +168,13 @@
             <version>${mockito.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <!-- Configures mockito to allow mocking of final classes and other features. -->
```
Comment (christrotteradhoc) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml:172]: Can you make a note here as well that this was added to resolve the illegal reflection warning?

I'm curious as well as to how/why this resolves it. I did a search and I see other references to this but I didn't really figure out what it is doing that fixes it. Anything found from your side that answers that question?

Comment (KyRobbins) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml:172]: If you look in the `mockito-inline` jar, it's just a couple ascii files, as the only thing it's doing is providing a one-line configuration to tell it which MockMaker implementation to use.  It's the difference of it using the `ByteBuddyMockMaker`, which results in the illegal reflection issue, versus the `InlineByteBuddyMockMaker` which doesn't, but as to how the implementations differ to cause the illegal reflection access or not, I'm not exactly sure.

https://github.com/hustwei/mockito/blob/master/src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java

In this PR's code's case, it was when Mockito changed the access level for the `protected volatile InputStream in` property of `FilterInputStream` that was throwing the illegal reflection.  Using the inline version, it's able to do this without causing an illegal reflection action.

Comment (KyRobbins) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml:172]: oh and I'll add the extra note

Comment (christrotteradhoc) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml:172]: Thanks. This is helpful. Would you mind putting this additional information in the PR description as well?

**Comment by christrotteradhoc** [comment]
Looks ready for merge to me pending @StewGoin review

**Review by StewGoin — APPROVED** [review]
No concerns from my end as ISSO.

**Review by christrotteradhoc — APPROVED** [review]

**Review by brianburton — APPROVED** [review]

**Review by lsmitchell — APPROVED** [review]

**Review by ericbuckley — APPROVED** [review]

**Pull request merged**